src/abi_arm.cpp

```diff
@@ -0,0 +1,304 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+
+//===----------------------------------------------------------------------===//
+//
+// The ABI implementation used for ARM targets.
+//
+//===----------------------------------------------------------------------===//
+//
+// The Procedure Call Standard can be found here:
+// http://infocenter.arm.com/help/topic/com.arm.doc.ihi0042f/IHI0042F_aapcs.pdf
+//
+//===----------------------------------------------------------------------===//
+
+// LIMITATIONS
+// - doesn't respect register usage, these should be encoded in the ABI state
```

Comment:
I think you're right, I had thought the split register/stack mode would need to be generated by the front-end but it seems part of ARMCallingConv.td.